lib/child_process.js

```diff
@@ -323,6 +324,11 @@ function normalizeSpawnArguments(file /*, args, options*/) {
   // Make a shallow copy so we don't clobber the user's options object.
   options = Object.assign({}, options);
 
+  if (options.shell === 'auto') {
```

Comment:
I considered that at first, but since the statement can make `options.shell` become `false`, that would mean that you'd have to break out of the `if` afterwards.